src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.cs

```diff
@@ -0,0 +1,2718 @@
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Plus_System_Single()
+        {
+            string source = @"
+class A
+{
+    System.Single Method()
+    {
+        System.Single i = default(System.Single);
+        return /*<bind>*/+i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.FloatingPlus) (OperationKind.UnaryOperatorExpression, Type: System.Single)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Single)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Plus_System_Double()
+        {
+            string source = @"
+class A
+{
+    System.Double Method()
+    {
+        System.Double i = default(System.Double);
+        return /*<bind>*/+i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.FloatingPlus) (OperationKind.UnaryOperatorExpression, Type: System.Double)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Double)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Plus_System_Boolean()
+        {
+            string source = @"
+class A
+{
+    System.Boolean Method()
+    {
+        System.Boolean i = default(System.Boolean);
+        return /*<bind>*/+i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Boolean)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Plus_System_Object()
+        {
+            string source = @"
+class A
+{
+    System.Object Method()
+    {
+        System.Object i = default(System.Object);
+        return /*<bind>*/+i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Object)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Minus_System_SByte()
```

Comment:
Tests seem very comprehensive, and I see that you have covered all the possible combinations of the huge UnaryOperationKind enum 👍